packages/mdl-base/README.md

```diff
@@ -0,0 +1,241 @@
+    button.type = 'button';
+    button.classList.add(BUTTON);
+    root.appendChild(button);
+
+    return root;
+  }
+
+  static attachTo(root) {
+    return new MyComponent(root);
+  }
+
+  getDefaultFoundation() {
+    const btn = this.root.querySelector(`.${MyComponentFoundation.cssClasses.BUTTON}`);
+    return new MyComponentFoundation({
+      toggleClass: className => {
+        if (this.root.classList.contains(className)) {
+          this.root.classList.remove(className);
+          return;
+        }
+        this.root.classList.add(className);
+      },
+      registerBtnClickHandler: handler => btn.addEventListener('click', handler),
+      deregisterBtnClickHandler: handler => btn.removeEventListener('click', handler)
+    });
+  }
+}
+```
+
+#### Properties
+
+`MDLComponent` provides the following "private" properties to subclasses:
+
+| property | description |
+| --- | --- |
+| `root_` | The root element passed into the constructor as the first argument. |
+| `foundation_` | The foundation class for this component. This is either passed in as an optional second argument to the constructor, or assigned the result of calling `getDefaultFoundation()` |
+
+#### Methods
+
+`MDLComponent` provides the following methods to subclasses:
+
+| method | description |
+| --- | --- |
+| `getDefaultFoundation()` | Returns an instance of a foundation class properly configured for the component. Called when no foundation instance is given within the constructor. Subclasses **must** implement this method. |
+| `initialSyncWithDOM()` | Called within the constructor. Subclasses may override this method if they wish to perform initial synchronization of state with the host DOM element. For example, a slider may want to check if its host element contains a pre-set value, and adjust its internal state accordingly. Note that the same caveats apply to this method as to foundation class lifecycle methods. Defaults to a no-op. |
+| `destroy()` | Subclasses may override this method if they wish to perform any additional cleanup work when a component is destroyed. For example, a component may want to deregister a window resize listener. |
+
+#### Static Methods
+
+In addition to methods inherited, subclasses should implement the following two static methods within their code:
+
+| method | description |
+| --- | --- |
+| `buildDom(...any) => HTMLElement` | Subclasses may implement this as a convenience method to construct the proper DOM for a component. Users could then rely on this as an alternative to having to construct the DOM themselves. However, it should exist purely for convenience and _never_ be used as a dependency for the component itself. |
+| `attachTo(root) => <ComponentClass>` | Subclasses must implement this as a convenience method to instantiate and return an instance of the class using the root element provided. This will be used within `mdl-auto-init`, and in the future its presence may be enforced via a custom lint rule.|
```

Comment:
I think so, because packages like `mdl-auto-init` as well as users can use it as a way of calling the constructor on an element as a side effect. So in this way it's kind of just sugar.